### ISSUE TITLE
Clippy fails to compile my code but cargo check succeeds

### ISSUE THREAD

**Issue by nicolaiunrein** [rust-lang/rust-clippy#11585]
### Summary

running cargo clippy --fix I get the following output:
```
warning: failed to automatically apply fixes suggested by rustc to crate `ltc_receiver`

after fixes were automatically applied the compiler reported errors within these files:

  * audio\ltc-receiver\src\main.rs
  * audio\ltc-receiver\src\state.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0631]: type mismatch in function arguments
    --> audio\ltc-receiver\src\state.rs:160:30
     |
160  |                     cfgs.any(config_is_sufficient)
     |                          --- ^^^^^^^^^^^^^^^^^^^^ expected due to this
     |                          |
     |                          required by a bound introduced by this call
     |
    ::: audio\ltc-receiver\src\main.rs:38:1
     |
38   | fn config_is_sufficient(c: &SupportedStreamConfigRange) -> bool {
     | --------------------------------------------------------------- found signature defined here
     |
     = note: expected function signature `fn(cpal_dt::SupportedStreamConfigRange) -> _`
                found function signature `for<'a> fn(&'a cpal_dt::SupportedStreamConfigRange) -> _`
note: required by a bound in `std::iter::Iterator::any`
    --> C:\Users\unrein\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\iter\traits\iterator.rs:2699:12
     |
2696 |     fn any<F>(&mut self, f: F) -> bool
     |        --- required by a bound in this associated function
...
2699 |         F: FnMut(Self::Item) -> bool,
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Iterator::any`
help: do not borrow the argument
    -->  audio\ltc-receiver\src\main.rs:38:28

38   - fn config_is_sufficient(c: &SupportedStreamConfigRange) -> bool {
38   + fn config_is_sufficient(c: SupportedStreamConfigRange) -> bool {
     |

error: aborting due to previous error

For more information about this error, try `rustc --explain E0631`.
Original diagnostics will follow.

warning: function `find_device` is never used
  --> audio\ltc-receiver\src\main.rs:26:4
   |
26 | fn find_device(host: &Host, name: &str) -> BoxResult<Device> {
   |    ^^^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: called `is_some()` after searching an `Iterator` with `find`
   --> audio\ltc-receiver\src\state.rs:160:26
    |
160 |                     cfgs.find(config_is_sufficient).is_some()
    |                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `any()` instead: `any(config_is_sufficient)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#search_is_some
    = note: `#[warn(clippy::search_is_some)]` on by default

warning: `ltc-receiver` (bin "ltc-receiver") generated 2 warnings (2 duplicates)
warning: `ltc-receiver` (bin "ltc-receiver" test) generated 2 warnings (run `cargo clippy --fix --bin "ltc-receiver" --tests` to apply 1 suggestion)
    Finished dev [unoptimized + debuginfo] target(s) in 5.13s
```

### Reproducer

I tried this code:

```rust
<code>
```

I expected to see this happen:

Instead, this happened:

### Version

```text
rustc 1.72.0 (5680fa18f 2023-08-23)
binary: rustc
commit-hash: 5680fa18feaa87f3ff04063800aec256c3d4b4be
commit-date: 2023-08-23
host: x86_64-pc-windows-msvc
release: 1.72.0
LLVM version: 16.0.5
```

**Comment by samueltardieu** [comment]
Closing as stale without reproducer.
